specification/search/data-plane/Azure.Search/preview/2025-05-01-preview/searchservice.json

```diff
@@ -0,0 +1,14715 @@
+        }
+      },
+      "description": "Contains configuration options related to vector search."
+    },
+    "AssetProfile": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "externalDocs": {
+            "url": "https://learn.microsoft.com/rest/api/searchservice/Naming-rules"
+          },
+          "type": "string",
+          "description": "The name to associate with this particular asset profile.",
+          "x-nullable": false
+        },
+        "assetType": {
+          "type": "string",
+          "enum": [
+            "path"
+          ],
+          "x-ms-enum": {
+            "name": "AssetType",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "path",
+                "name": "Path",
+                "description": "The part of the Url path after the first compoenet, which is determined by the container name configured in asset store."
+              }
+            ]
+          },
+          "description": "The asset type determines how to locate an asset in the asset store."
+        },
+        "servingType": {
+          "type": "string",
+          "enum": [
+            "sasUrl"
+          ],
+          "x-ms-enum": {
+            "name": "ServingType",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "sasUrl",
+                "name": "SasUrl",
+                "description": "A secure URL that grants restricted access to a resource, allowing users to perform specific actions for a limited period without exposing the account key."
+              }
+            ]
+          },
+          "description": "The serving type determines how to serve assets to API callers."
+        },
+        "servingOptions": {
+          "type": "object",
+          "properties": {
+            "sasExpiry": {
+              "type": "string",
+              "pattern": "^P(?=\\d|T\\d)(\\d+Y)?(\\d+M)?(\\d+W)?(\\d+D)?(T(\\d+H)?(\\d+M)?(\\d+S)?)?$",
+              "example": "PT1H",
+              "default": "PT1H",
+              "description": "Specifies the duration for which the generated SAS URL will remain valid, formatted as an ISO 8601 duration."
+            }
+          }
+        },
+        "assetStore": {
+          "$ref": "#/definitions/SearchAssetStore",
+          "description": "The definition of an asset store that stores the assets."
+        }
+      }
+    },
+    "SearchAssetStoreType": {
+      "type": "string",
+      "enum": [
+        "azureblob"
+      ],
+      "x-ms-enum": {
+        "name": "SearchAssetStoreType",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "azureblob",
+            "name": "AzureBlob",
+            "description": "Indicates an Azure Blob asset store."
+          }
+        ]
+      },
+      "description": "Defines the type of an asset store."
+    },
+    "AssetStoreCredentials": {
```

Comment:
There are some naming inconsistencies going on with this feature. Some are prefixed with `Search` and others aren't, should this be made consistent?